### PR TITLE
Improve docs (readme and doxygen)

### PR DESCRIPTION
**Pull request by bdilly**
Signed-off-by: Bruno Dilly <bruno.dilly@intel.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -136,21 +156,45 @@ or stopped as well as monitored for a dynamic lifecycle
...
+##### COAP
 
 Implementation of The Constrained Application Protocol (CoAP -
 RFC 7252). This network protocol is pretty simple and uses a
 HTTP-like message that is space-efficient over UDP.
 
+##### OIC
+
+Implementation of protocol defined by Open Interconnect Consortium
+(OIC - http://openinterconnect.org/)
```
Comment (lpereira) [inline README.md]: Use a link here: [Open Interconnect Consortium (OIC)](http://openinterconnect.org).

File: README.md [hunk 2]
```diff
@@ -190,7 +234,7 @@ sol_fd_add() and sol_fd_del() are thread-safe and can be used from
 worker threads.
 
 
-##### GLib (kconfig: Core library -> Mainloop -> glib)
+#### GLib (kconfig: Core library -> Mainloop -> glib)
```
Comment (lpereira) [inline README.md]: Maybe use `→` instead of `->`?

File: README.md (hunk 1 above)
Comment (bdilly) [inline README.md]: It creates links automagically:
Please check it on https://github.com/bdilly/soletta/

File: README.md [hunk 3]
```diff
@@ -136,21 +156,45 @@ or stopped as well as monitored for a dynamic lifecycle
...
+
+##### Network
+
+Network library provides a way to handle network link interfaces,
+making it possible to observe events, to inquire available links
+and to set their states.
+
+##### COAP
 
 Implementation of The Constrained Application Protocol (CoAP -
```
Comment (lpereira) [inline README.md]: Maybe link to the RFC here?

File: README.md [hunk 4]
```diff
@@ -273,7 +317,7 @@ A set of services are started automatically by Soletta in order to do
 initialization, these are listed in
 /usr/lib/soletta/modules/linux-micro/initial-services.
 
-### Flow Based Programming
+## Flow Based Programming
 
 Flow-Based Programming (**FBP**) allows the programmer to express business
```
Comment (lpereira) [inline README.md]: A link to Wikipedia might help as a starting point for the reader to know more about FBP

File: README.md [hunk 5]
```diff
@@ -243,7 +287,7 @@ Thus both targets and services are implemented according to systemd
 using units in standard locations such as /usr/lib/systemd/system and
 /etc/systemd/system.
```
Comment (lpereira) [inline README.md]: For paths, commands, and constants, I like to emphasize them by quoting them in backticks. GitHub renders this with a fixed-width font and light gray background.

File: README.md [hunk 6]
```diff
@@ -208,13 +252,13 @@ in-depth information, the summary of environment variables to use:
 
 	make CFLAGS="-O0 -ggdb3" # disable optimizations
 
-##### POSIX (kconfig: Core library -> Mainloop -> posix)
+#### POSIX (kconfig: Core library -> Mainloop -> posix)
 
 If GLib is too big then you can use a simpler implementation based
```
Comment (lpereira) [inline README.md]: Link to GLib?

File: README.md [hunk 7]
```diff
@@ -208,13 +252,13 @@ in-depth information, the summary of environment variables to use:
...
 solely on POSIX syscalls (poll/ppoll). As it's fully implemented in
```
Comment (lpereira) [inline README.md]: `poll(2)/ppoll(2)`?

File: README.md [hunk 8]
```diff
@@ -208,13 +252,13 @@ in-depth information, the summary of environment variables to use:
...
 soletta there is no extra variables to debug it.
```
Comment (lpereira) [inline README.md]: Capitalize the project name.

File: README.md (hunk 1 above)
Comment (bdilly) [inline README.md]: I think I got your idea now.
Nvm.

File: README.md [hunk 9]
```diff
@@ -136,21 +156,45 @@ or stopped as well as monitored for a dynamic lifecycle
...
+technologies to wirelessly connect and intelligently manage
+the flow of information among devices, regardless of form factor,
+operating system or service provider.
+
+Both client and server sides are covered by this library.
 
-##### flow
+#### flow
 
 Implementation of Flow-Based Programming (FBP), allowing the
```
Comment (lpereira) [inline README.md]: Maybe a link to the larger FBP section down below?

File: README.md [hunk 10]
```diff
@@ -136,21 +156,45 @@ or stopped as well as monitored for a dynamic lifecycle
...
 
+##### OIC
+
+Implementation of protocol defined by Open Interconnect Consortium
+(OIC - http://openinterconnect.org/)
+
+It's a common communication framework based on industry standard
+technologies to wirelessly connect and intelligently manage
+the flow of information among devices, regardless of form factor,
+operating system or service provider.
```
Comment (lpereira) [inline README.md]: Nitpick: use Oxford comma?

File: README.md [hunk 11]
```diff
@@ -136,21 +156,45 @@ or stopped as well as monitored for a dynamic lifecycle
 management.
 
 
-##### coap
+#### comms
+
+Comms consists on a few communication modules.
+It provides ways to deal with network, CoAP protocol and
```
Comment (lpereira) [inline README.md]: CoAP and OIC should be links to their respective sections down below.

**Comment by bdilly** [comment]
OK, I'll make some of these changes on this commit.
Some other are not related to the changes done, but are aligned with the spirit of improving docs.
I'm going to review these points and send a v2 soon.
Thanks

File: README.md [hunk 12]
```diff
@@ -13,8 +13,28 @@ Portable and scalable, it abstracts details of hardware and OS,
...
+ * [Platforms](#platforms)
+  * [Systemd](#systemd-kconfig-core-library---target-platform---systemd)
+  * [Linux-micro](#linux-micro-kconfig-core-library---target-platform---linux-micro)
+ * [Flow Based Programming](#flow-based-programming)
+ * [Contributing](#contributing)
+
+
+## General Information
 
 **Soletta Project** uses `sol` as C namespace, so macros start with `SOL_`
```
Comment (lpereira) [inline README.md]: I'm not sure if this should be in the "General Information" section. In fact, this section deals with the building process more than anything, so it should be properly named.

File: README.md [hunk 13]
```diff
@@ -243,7 +287,7 @@ Thus both targets and services are implemented according to systemd
 using units in standard locations such as /usr/lib/systemd/system and
 /etc/systemd/system.
 
-##### Linux-micro (kconfig: Core library -> Target Platform -> linux-micro)
+#### Linux-micro (kconfig: Core library -> Target Platform -> linux-micro)
 
 Linux-micro implementation allows your Soletta binary to be used as
```
Comment (lpereira) [inline README.md]: Avoid using "your"/"our". Use "a"/"an" instead.

File: README.md (hunk 12 above)
Comment (bdilly) [inline README.md]: I'm going to split it in two sections, adding a specific about "building from source"

**Comment by bdilly** [comment]
Merged with most of suggestions applied.
I'm not adopting oxford commas and fancy arrows.

**Pull request closed without merging**